containers/src/Kokkos_DualView.hpp

```diff
@@ -925,6 +961,19 @@ class DualView : public ViewTraits<DataType, Arg1Type, Arg2Type, Arg3Type> {
       modified_flags(1) = modified_flags(0) = 0;
```

Comment:
Not a behavior change in this PR that I can tell, but could @stanmoore1 or someone else familiar with `DualView` semantics check that in the case where size is the same and no reallocation happens, and we do initialize *only on the device instance*, that the resulting `modified_flags` values are sensible? It seems like a subsequent synchronization call should recognize that the re-initialized data on the device side is fresher than the stale data on the host side.